caput/memh5.py

```diff
@@ -2152,40 +2187,89 @@ def _map_json(value):
 
 
 def deep_group_copy(
-    g1, g2, convert_dataset_strings=False, convert_attribute_strings=False
+    g1,
+    g2,
+    selections=None,
+    convert_dataset_strings=False,
+    convert_attribute_strings=False,
 ):
-    """Copy full data tree from one group to another."""
+    """
+    Copy full data tree from one group to another.
+
+    Copies from g1 to g2. An axis downselection can be specified by supplying the
+    parameter 'selections'. If for example 'g1' is
+
+    ```
+    foo:
```

Comment:
Can you write this example in the standard `doctest` format? If you do the example should automatically get run and tested in the CI run.